src/test/kotlin/com/example/Employee/controllers/BankAccountControllerTest.kt

```diff
@@ -0,0 +1,68 @@
+package com.example.Employee.controllers
+
+import com.example.Employee.controllers.request.BankAccountRequest
+import com.example.Employee.models.BankAccount
+import com.example.Employee.models.Employee
+import com.example.Employee.services.BankAccountService
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.jupiter.api.Test
+import org.amshove.kluent.`should be equal to`
+
+internal class BankAccountControllerTest {
+    private val bankAccountService=mockk<BankAccountService>()
+    private val bankAccountController=BankAccountController(bankAccountService)
+
+    @Test
+    fun `should add bank details of employee`(){
+        every { bankAccountService.addAccount(1, fakeBankAccountRequest) }returns Unit
+
+        val addedAccount=bankAccountController.addAccount(1, fakeBankAccountRequest)
+        println(addedAccount)
+
+        addedAccount.body `should be equal to` fakeBankAccountRequest
+    }
+
+    @Test
+    fun `should get all bank accounts of an employee`(){
+        every { bankAccountService.getAccounts(1) }returns listOf(fakeBankAccount)
+
+        val accountsFetched=bankAccountController.getAccounts(1)
+
+        accountsFetched.body `should be equal to` listOf(fakeBankAccount)
+    }
+
+    @Test
+    fun `should get a particular bank account of an employee`(){
+        every { bankAccountService.getAccount(1,1) }returns fakeBankAccount
+
+        val accountFetched= bankAccountController.getAccount(1,1)
+
+        accountFetched.body `should be equal to` fakeBankAccount
+    }
+
+    @Test
+    fun `should update a bank account`(){
+        every { bankAccountService.updateAccount(1,1, fakeBankAccountRequest.copy(bankName = "HDFC")) }returns Unit
+
+        val updatedAccount = bankAccountController.updateAccount(1,1, fakeBankAccountRequest.copy(bankName = "HDFC"))
+
+        updatedAccount.body `should be equal to` fakeBankAccountRequest.copy(bankName = "HDFC")
+    }
+
+    @Test
+    fun `should delete a bank account`(){
+        every { bankAccountService.deleteAccount(1,1) }returns Unit
+
+        bankAccountController.deleteAccount(1,1)
```

Comment:
Verify the response as well.

Reply:
I actually didn’t write any response for delete requests